app/models/users/setting.rb

```diff
@@ -0,0 +1,41 @@
+module Users
+  class Setting < ApplicationRecord
+    self.table_name_prefix = "users_"
+
+    belongs_to :user
+
+    enum editor_version: { v2: 0, v1: 1 }, _suffix: :editor
+    enum config_font: { default: 0, comic_sans: 1, monospace: 2, open_dyslexic: 3, sans_serif: 4, serif: 5 }
+    enum inbox_type: { private: 0, open: 1 }, _suffix: :inbox
+    enum config_navbar: { default_navbar: 0, static_navbar: 1 }
+    enum config_theme: { default_theme: 0, minimal_light_theme: 1, night_theme: 2, pink_theme: 3,
+                         ten_x_hacker_theme: 4 }
+
+    before_validation :normalize_config_values
```

Comment:
I have an observation/question:

While working on [this spec](spec/models/users/setting_spec.rb:38), I thought I could set `setting.config_theme = "night theme"`, even though `config_theme` is an enum.

In the current `setting.rb` file, the method `normalize_config_values` is called before validations, and it replaces spaced in the config value to underscores.

What I am not sure of is whether this method runs before the enum validations. I ask because when the spec read:

```ruby
it "accepts valid theme" do
        setting.config_theme = 'night theme'
        expect(setting.night_theme?).to be true
end
``` 
the spec gave an error: `ArgumentError: 'night theme' is not a valid config_theme`. I am not sure if it is due to our use of enums, or that the `normalize_config_values` method is not called before the enum validation.

Thanks for helping/explaining 😅

Reply:
Tbh, I would probably just delete `normalize_config_values` and the `before_validation` and update the forms instead to send the correct values. This is data that is completely under our control, so sending it in a way that still needs processing seems a bit odd to me.

Reply:
I agree with @citizen428, let's remove this preoccupation from our minds. As we control where we set it, we can simply set it to the correct value 🙌